packages/bpk-component-infinite-scroll/src/withInfiniteScroll.js

```diff
@@ -113,108 +114,111 @@ const withInfiniteScroll = (
         showSeeMore: false,
       };
 
-      this.onItemsFetch = props.onItemsFetch;
-      this.observer = new IntersectionObserver(this.handleIntersection, {
-        threshold: 1,
-      });
-    }
-
-    componentDidMount() {
-      if (this.sentinel) {
-        this.observer.observe(this.sentinel);
-      }
+      this.props.dataSource.onDataChange(this.updateData);
     }
 
-    componentDidUpdate() {
-      if (this.sentinel && this.state.index > 0) {
-        this.observer.observe(this.sentinel);
-      }
+    componentWillMount() {
```

Comment:
We should use `componentDidMount`. `componentWillMount` has been removed from the new versions and it can be potentially called more than once.